src/pages/home/live-pricing/components/_live_market_table.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import {
+    flexRender,
+    getCoreRowModel,
+    getSortedRowModel,
+    useReactTable,
+} from '@tanstack/react-table'
+import type { SortingState } from '@tanstack/react-table'
+import { TAvailableLiveMarkets, TMarketData, TMarketDataResponse } from '../_types'
+import useLiveColumns from '../_use-live-columns'
+import { TABLE_VISIBLE_ROWS } from '../_utils'
+import Spinner, { TableLoadingContainer, Table, TableContainer, TableRow } from './_elements'
+import { useDerivApi } from 'components/hooks/use-deriv-api'
+
+export type TLiveMarketTableProps = {
+    market: TAvailableLiveMarkets
+}
+
+const LiveMarketTable = ({ market }: TLiveMarketTableProps) => {
+    const [markets_data, setMarketsData] = useState(() => {
+        const temp = new Map<TAvailableLiveMarkets, TMarketData[]>()
+        return temp
+    })
+
+    const [is_loading, setIsLoading] = useState(false)
+
+    const table_data = useMemo(() => {
+        const data = markets_data.get(market)
+
+        return data ?? []
+    }, [market, markets_data])
+
+    const [sorting, setSorting] = React.useState<SortingState>([])
+
+    const { send } = useDerivApi()
+
+    const requestMarketsData = useCallback(() => {
+        setIsLoading(true)
+
+        send(
+            { trading_platform_asset_listing: 1, platform: 'mt5', type: 'brief' },
+            (response: TMarketDataResponse) => {
+                const responseData = [...response.trading_platform_asset_listing.mt5.assets]
+                const markets = new Map<TAvailableLiveMarkets, TMarketData[]>()
+
+                responseData.forEach((item) => {
+                    const currentMarket = [...(markets.get(item.market) ?? [])]
```

Comment:
Seems it always should return empty due the markets is empty from beginning
BTW for workable function we don't need `?? []`